dora/core/client/fs/src/main/java/alluxio/client/file/dora/DoraCachePositionReader.java

```diff
@@ -74,4 +80,15 @@ public synchronized void close() throws IOException {
     mNettyReader.close();
     mFallbackReader.close();
   }
+
+  private int fallback(long position, ReadTargetBuffer buffer,
+      int length, int originalOffset) throws IOException {
+    // In case any error in worker read, revert the offset change in the buffer
+    buffer.offset(originalOffset);
```

Comment:
this can be set by the caller before calling this method.

Reply:
Do you mean make the fallback function as a parameter of this method?  I think it's really a good idea.

Reply:
Sorry that was not clear. I meant something like this:

```java
private int fallback(long position, ReadTargetBuffer buffer,
      int length) throws IOException {
  int read = mFallbackReader.get().read(position, buffer, length);
  ...
}

// caller
buffer.offset(originalOffset);
return fallback(position, buffer, length);
```